src/cdk/table/table.ts

```diff
@@ -443,14 +443,18 @@ export class CdkTable<T> implements AfterContentChecked, CollectionViewer, OnDes
   constructor(
       protected readonly _differs: IterableDiffers,
       protected readonly _changeDetectorRef: ChangeDetectorRef,
-      protected readonly _coalescedStyleScheduler: _CoalescedStyleScheduler,
       protected readonly _elementRef: ElementRef, @Attribute('role') role: string,
       @Optional() protected readonly _dir: Directionality, @Inject(DOCUMENT) _document: any,
       private _platform: Platform,
-      // Optional for backwards compatibility, but a view repeater strategy will always
-      // be provided.
+
+      /**
+       * @deprecated `_coalescedStyleScheduler` and `_viewRepeater` parameters to become required.
+       * @breaking-change 11.0.0
+       */
       @Optional() @Inject(_VIEW_REPEATER_STRATEGY)
-      protected readonly _viewRepeater: _ViewRepeater<T, RenderRow<T>, RowContext<T>>) {
+        protected readonly _viewRepeater?: _ViewRepeater<T, RenderRow<T>, RowContext<T>>,
+      @Optional() @Inject(_COALESCED_STYLE_SCHEDULER)
```

Comment:
I'm not totally sure whether the lightweight token makes sense here since the base class always provides a style scheduler which means that the code will always be retained. I added one since the lint rule flagged it. cc @devversion

Reply:
Did it flag the parameter because you had `@Optional` on it initially? I think as per [this discussion](https://github.com/angular/components/pull/19920#discussion_r452181383) it shouldn't have that. I can't see the diff on how it was before; so maybe the code was different.

Generally, I think it _could_ makes sense if one extends the class and provides an explicit constructor where he doesn't want to use the coalesced style scheduler (and now doesn't have to pay the size cost of it).

Once this becomes required, the lightweight token doesn't make any sense anymore though, but we probably can persist it then to avoid another breaking change? Could also see us doing a lint rule exception for this.

Reply:
I see now, I forgot that we had a special case for `@breaking-change`. It probably didn't work, because I used a single comment for `_viewRepeater` and `_coalescedStyleScheduler`.

Reply:
Yeah, the breaking change logic we removed though since you made the point that parameters using `@breaking-change` and are "to be made required" shouldn't be `@Optional` in the first place. 

And sounds good. I think it doesn't matter a lot 👍